x-pack/plugins/fleet/server/routes/settings/enrollment_settings_handler.ts

```diff
@@ -0,0 +1,185 @@
+      await getFleetServerAgentPolicies(soClient, agentPolicyId);
+    const scopedAgentPolicy = scopedAgentPolicyResponse || {
+      id: undefined,
+      name: undefined,
+      fleet_server_host_id: undefined,
+      download_source_id: undefined,
+    };
+
+    // Check if there is any active fleet server enrolled into the fleet server policies policies
+    if (fleetServerAgentPolicies) {
+      settingsResponse.fleet_server.agent_policies = fleetServerAgentPolicies;
+      settingsResponse.fleet_server.has_active = await hasActiveFleetServersForAgentPolicies(
+        esClient,
+        soClient,
+        fleetServerAgentPolicies.map((p) => p.id)
+      );
+    }
+
+    // Get download source
+    // ignore errors if the download source is not found
+    try {
+      settingsResponse.download_source = await getDownloadSource(
+        soClient,
+        scopedAgentPolicy.download_source_id ?? undefined
+      );
+    } catch (e) {
+      settingsResponse.download_source = undefined;
+    }
+
+    // Get associated fleet server host, or default one if it doesn't exist
+    // `getFleetServerHostsForAgentPolicy` errors if there is no default, so catch it
+    try {
+      settingsResponse.fleet_server.host = await getFleetServerHostsForAgentPolicy(
+        soClient,
+        scopedAgentPolicy
+      );
+    } catch (e) {
+      settingsResponse.fleet_server.host = undefined;
+    }
+
+    // if a fleet server host was found, get associated fleet server host proxy if any
+    // ignore errors if the proxy is not found
+    try {
+      if (settingsResponse.fleet_server.host?.proxy_id) {
+        settingsResponse.fleet_server.host_proxy = await getFleetProxy(
+          soClient,
+          settingsResponse.fleet_server.host.proxy_id
+        );
+      }
+    } catch (e) {
+      settingsResponse.fleet_server.host_proxy = undefined;
+    }
+
+    return response.ok({ body: settingsResponse });
+  } catch (error) {
+    return defaultFleetErrorHandler({ error, response });
+  }
+};
+
+const getFleetServerAgentPolicies = async (
+  soClient: SavedObjectsClientContract,
+  agentPolicyId?: string
+): Promise<{
+  fleetServerAgentPolicies?: EnrollmentSettingsAgentPolicy[];
+  scopedAgentPolicy?: EnrollmentSettingsAgentPolicy;
+}> => {
+  const mapPolicy = (policy: AgentPolicy) => ({
+    id: policy.id,
+    name: policy.name,
+    is_managed: policy.is_managed,
+    is_default_fleet_server: policy.is_default_fleet_server,
+    has_fleet_server: policy.has_fleet_server,
+    fleet_server_host_id: policy.fleet_server_host_id,
+    download_source_id: policy.download_source_id,
+  });
+
+  // If an agent policy is specified, return only that policy
+  if (agentPolicyId) {
+    const agentPolicy = await agentPolicyService.get(soClient, agentPolicyId, true);
+    if (agentPolicy) {
+      if (agentPolicy.package_policies?.find((p) => p.package?.name === FLEET_SERVER_PACKAGE)) {
```

Comment:
I thought that `scopedAgentPolicy` just returns the policy with the requested id, but with this check it won't be returned if that policy has no `fleet-server` integration. Do we need it?

Reply:
yes, if the `scopedAgentPolicy` is not a fleet server policy, the rest of the information is still tailored for that agent policy: the fleet server host to use, its proxy, and download source